app/Http/Requests/SignupRequest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class SignupRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules()
+    {
+        return [
+            'email' => 'required',
+            'username' => 'required',
+            'first_name' => 'required',
+            'last_name' => 'required',
+            'password' => 'required',
```

Comment:
tương tự bên dưới này, bổ sung điều kiện validate cho m nhé, thế này hơi sơ sài quá, nó ko an toàn